mikrotik/resource_interface_wireguard_peer.go

```diff
@@ -0,0 +1,264 @@
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+				Description: "List of IP (v4 or v6) addresses with CIDR masks from which incoming traffic for this peer is allowed and to which outgoing traffic for this peer is directed. The catch-all 0.0.0.0/0 may be specified for matching all IPv4 addresses, and ::/0 may be specified for matching all IPv6 addresses.",
+			},
+			"comment": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+				Description: "Short description of the peer.",
+			},
+			"disabled": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+				Description: "Boolean for whether or not the interface peer is disabled.",
+			},
+			"endpoint_address": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+				Description: "An endpoint IP or hostname can be left blank to allow remote connection from any address.",
+			},
+			"endpoint_port": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+				Default:  int64default.StaticInt64(0),
+				Validators: []validator.Int64{
+					int64validator.Between(0, 65535),
+				},
+				Description: "An endpoint port can be left blank to allow remote connection from any port.",
+			},
+			"interface": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the WireGuard interface the peer belongs to.",
+			},
+			"persistent_keepalive": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+				Default:  int64default.StaticInt64(0),
+				Validators: []validator.Int64{
+					int64validator.Between(0, 65535),
+				},
+				Description: "A seconds interval, between 1 and 65535 inclusive, of how often to send an authenticated empty packet to the peer for the purpose of keeping a stateful firewall or NAT mapping valid persistently. For example, if the interface very rarely sends traffic, but it might at anytime receive traffic from a peer, and it is behind NAT, the interface might benefit from having a persistent keepalive interval of 25 seconds.",
+			},
+			"preshared_key": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+				Description: "A base64 preshared key. Optional, and may be omitted. This option adds an additional layer of symmetric-key cryptography to be mixed into the already existing public-key cryptography, for post-quantum resistance.",
+			},
+			"public_key": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The remote peer's calculated public key.",
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (i *interfaceWireguardPeer) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
```

Comment:
Can you add a TODO comment here to reference that this should be migrated to the newer generic crud interface? A good example of that is the [bgp_peer resource](https://github.com/ddelnano/terraform-provider-mikrotik/blob/095ab71bd699526cda3ff32ac8dab3f7d97ac338/mikrotik/resource_bgp_peer.go#L204-L229).

I attempted that change for this resource, but terraform was failing with the following error and I didn't have a quick fix for it. 
```
=== CONT  TestAccMikrotikInterfaceWireguardPeer_create
    resource_interface_wireguard_peer_test.go:22: Step 1/1 error: Error running apply: exit status 1

        Error: Value Conversion Error

          with mikrotik_interface_wireguard_peer.bar,
          on terraform_plugin_test.tf line 9, in resource "mikrotik_interface_wireguard_peer" "bar":
           9:   resource "mikrotik_interface_wireguard_peer" "bar" {

        An unexpected error was encountered trying to convert tftypes.Value into
        mikrotik.interfaceWireguardPeer. This is always an error in the provider.
        Please report the following to the provider developer:

        mismatch between struct and object: Object defines fields not found in
        struct: endpoint_address, endpoint_port, id, public_key, allowed_address,
        comment, persistent_keepalive, disabled, preshared_key, and interface.
```

Reply:
I don't want that to prevent this PR from merging, but the new interface is significantly less code.

Reply:
Yeah, definitely looks like it's much simpler; I just added the comment.